src/App.tsx

```diff
@@ -17,18 +72,26 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                setSortType={setSortType}
+                setQuery={setQuery}
+                query={query}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {!isLoaded && (
+                <Loader />
+              )}
+              {isFirstRender ? (
+                <TodoList todos={todos} />
+              ) : (
+                <TodoList todos={filteredTodos} />
+              )}
```

Comment:
You don't need this logic
```suggestion
                <TodoList todos={filteredTodos} />
```